### PR TITLE
Add BaseJob status methods

### PR DESCRIPTION
**Pull request by luisg5**
### Summary

Add the `done()`, `running()`, `cancelled()`, and `in_final_state()` methods to `BaseJob`.

### Details and comments

The implementation of the first three methods were initially added to `IBMQJob` in the provider.
As discussed in https://github.com/Qiskit/qiskit-ibmq-provider/issues/491, it might be better to add these methods to `BaseJob` so that all jobs can benefit from the methods implicitly. `in_final_state()` is added since it is also another convenient function.

### CODE REVIEW

**Comment by ewinston** [comment]
Looks good. Another convenient function is checking job.status() in JOB_FINAL_STATES.

File: qiskit/providers/basejob.py [hunk 1]
```diff
@@ -43,6 +45,18 @@ def backend(self):
         """Return the backend where this job was executed."""
         return self._backend
 
+    def done(self) -> bool:
+        """Return whether the job has successfully run."""
+        return self.status() == JobStatus.DONE
+
+    def running(self):
```
Comment (diego-plan9) [inline qiskit/providers/basejob.py]: Can you add the return type hint here and to `cancelled()`, for consistency?

Comment (luisg5) [reply qiskit/providers/basejob.py]: Definitely! That's a nice catch, thanks!

**Comment by luisg5** [comment]
> Looks good. Another convenient function is checking job.status() in JOB_FINAL_STATES.

Added an `in_final_state()` method as well. If there is a clearer or better name, then I could change it 😄

**Comment by jyu00** [comment]
> ... might be relevant if we still want to keep `BaseJob` as close as possible to the spec. 

@diego-plan9  During the last meeting we said we'd decouple `BaseJob` from the spec to allow quicker changes. So this one should be all good to go!

**Comment by jyu00** [comment]
@luisg5 Why does `Files changes` show a bunch of `qiskit_terra.dist-info` files? Is it just some Github anomaly?

**Comment by luisg5** [comment]
> Why does Files changes show a bunch of qiskit_terra.dist-info files? Is it just some Github anomaly?

They must have been added by accident. Removed them, since it's metadata generated by pip and it's not included in the project.

**Pull request merged**